clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -556,10 +557,15 @@ public static Set<TopicPartition> generateRandomTopicPartitions(int numTopic, in
      * @return The caught exception cause
      */
     public static <T extends Throwable> T assertFutureThrows(Future<?> future, Class<T> exceptionCauseClass) {
-        ExecutionException exception = assertThrows(ExecutionException.class, future::get);
-        assertInstanceOf(exceptionCauseClass, exception.getCause(),
-            "Unexpected exception cause " + exception.getCause());
-        return exceptionCauseClass.cast(exception.getCause());
+        try {
+            future.get(DEFAULT_MAX_WAIT_MS, TimeUnit.SECONDS);
+            fail("Future should throw expected exception " + exceptionCauseClass.getSimpleName() + " but succeed.");
+        } catch (TimeoutException | InterruptedException | ExecutionException e) {
+            assertInstanceOf(exceptionCauseClass, e.getCause(),
+                    "Unexpected exception cause " + e.getCause());
+            return exceptionCauseClass.cast(e.getCause());
+        }
+        return null;
```

Comment:
Looks better! Comments:
1. `DEFAULT_MAX_WAIT_MS` is a millisecond time unit, not seconds. Same comment applied to L583.
2. Now, if the future throws `TimeoutException` or `InterruptedException`, it will fail as expected. But what error message will we get? In your opinion, do you think that's helpful for troubleshooting?
3. What happened if the future throws `CancellationException`? return null? Is there any better way to handle it?

Thanks.